receiver/awss3receiver/s3reader.go

```diff
@@ -64,15 +67,45 @@ func (s3Reader *s3Reader) readAll(ctx context.Context, telemetryType string, dat
 	}
 
 	for currentTime := s3Reader.startTime; currentTime.Before(s3Reader.endTime); currentTime = currentTime.Add(timeStep) {
+		s3Reader.sendStatus(ctx, StatusNotification{
+			TelemetryType: telemetryType,
+			IngestStatus:  IngestStatusIngesting,
+			StartTime:     s3Reader.startTime,
+			EndTime:       s3Reader.endTime,
+			IngestTime:    currentTime,
+		})
+
 		select {
 		case <-ctx.Done():
+			s3Reader.sendStatus(ctx, StatusNotification{
+				TelemetryType: telemetryType,
+				IngestStatus:  IngestStatusCompleted,
```

Comment:
If the context finishes, would this be considered "completed" or "failed"? I suspect a context finish would indicate that something has gone wrong elsewhere and the transfer cannot be successfully completed.

Reply:
@adcharre This isn't critical since completed/failed are "done" either way, but what are your thoughts on this?

Reply:
I think this probably should be failed and also the return value should be the context error.